protocol-units/bridge/chains/movement/src/lib.rs

```diff
@@ -0,0 +1,198 @@
+use aptos_sdk::{move_types::language_storage::TypeTag, rest_client::Client, types::LocalAccount};
+use aptos_types::account_address::AccountAddress;
+use bridge_shared::{
+	bridge_contracts::{
+		BridgeContractCounterparty, BridgeContractCounterpartyError,
+		BridgeContractCounterpartyResult,
+	},
+	types::{
+		Amount, BridgeTransferDetails, BridgeTransferId, HashLock, HashLockPreImage,
+		InitiatorAddress, RecipientAddress, TimeLock,
+	},
+};
+use rand::prelude::*;
+use serde::Serialize;
+use std::str::FromStr;
+use std::sync::{Arc, Mutex};
+use url::Url;
+
+use crate::utils::MovementAddress;
+
+pub mod utils;
+
+const DUMMY_ADDRESS: AccountAddress = AccountAddress::new([0; 32]);
+const COUNTERPARTY_MODULE_NAME: &str = "atomic_bridge_counterparty";
+
+enum Call {
+	Lock,
+	Complete,
+	Abort,
+	GetDetails,
+}
+
+pub struct Config {
+	pub rpc_url: Option<String>,
+	pub ws_url: Option<String>,
+	pub chain_id: String,
+	pub signer_private_key: String,
+	pub initiator_contract: MovementAddress,
+	pub gas_limit: u64,
+}
+
+#[allow(dead_code)]
+#[derive(Clone)]
+pub struct MovementClient {
+	///Address of the counterparty moduke
+	counterparty_address: AccountAddress,
+	///Address of the initiator module
+	initiator_address: Vec<u8>,
+	///The Apotos Rest Client
+	rest_client: Client,
+	///The signer account
+	signer: Arc<LocalAccount>,
+}
+
+impl MovementClient {
+	pub async fn new(config: Config) -> Result<Self, anyhow::Error> {
```

Comment:
Given this is relying on random account generation, I don't think it should be called `new`.

Reply:
Agreed, it will be a `new_for_test` test feature flagged method. There's a need to merge this feature branch in to main soon. So I'm adding a small refactor tracking issue for this.

Reply:
Code intended for testing is dotted throughout, refactor issue created: https://github.com/movementlabsxyz/movement/issues/380